### PR TITLE
rename bian

### PR DESCRIPTION
**Pull request by SoluMilken**

### CODE REVIEW

File: uttut/pipeline/edit/base.py [hunk 1]
```diff
@@ -4,50 +4,50 @@
...
-class Edit(ABC):
+class Replacement(ABC):
     '''Represents an edit to a Sequence such as a string or a list
 
     A sequence is a collection of objects that can be indexed in a linear fashion. E.g. a string
 
-    A edit to a sequence is a replacement of a continuous range of the sequence with the elements
-    of the replacement sequence.
+    A edit to a sequence is a new_sequence of a continuous range of the sequence with the elements
+    of the new_sequence sequence.
```
Comment (absolutelyNoWarranty) [inline uttut/pipeline/edit/base.py]: 這句變好奇怪

File: uttut/pipeline/edit/base.py [hunk 2]
```diff
@@ -4,50 +4,50 @@
...
+    The transformation from "aBcdef" to "": Replacement(0, 6, "")
+    Note that insertion at position i is represented as Replacement(i, i, ...)
     ("Insertion at i" means that the original i-th element is moved to the right;
      similar to Python's list.insert)
 
     Args:
         start (int): start index (inclusive)
         end (int): end index (exclusive)
-        replacement (obj): the replacement
+        new_sequence (obj): the new_sequence
```
Comment (absolutelyNoWarranty) [inline uttut/pipeline/edit/base.py]: 應該是 new_subsequncence吧

File: uttut/pipeline/edit/base.py (hunk 1 above)
Comment (SoluMilken) [reply uttut/pipeline/edit/base.py]: A edit to a sequence is a new_sequence of a continuous range of the sequence with the elements
    of the new_sequence sequence.
要改成？
A replacement to a sequence is ????

File: uttut/pipeline/edit/base.py (hunk 2 above)
Comment (SoluMilken) [reply uttut/pipeline/edit/base.py]: 名字越來越長拉拉

**Comment by SoluMilken** [comment]
還是說把 bian改成 variation 就好？

File: uttut/pipeline/edit/base.py [hunk 3]
```diff
@@ -4,50 +4,50 @@
...
-class Edit(ABC):
+class Replacement(ABC):
     '''Represents an edit to a Sequence such as a string or a list
 
     A sequence is a collection of objects that can be indexed in a linear fashion. E.g. a string
 
-    A edit to a sequence is a replacement of a continuous range of the sequence with the elements
-    of the replacement sequence.
+    A replacement to a sequence is a new_value of a continuous range of the sequence with
+    the elements of the new_value sequence.
```
Comment (noobOriented) [inline uttut/pipeline/edit/base.py]: A replacement maps a slice of original sequence to a new value.

Comment (SoluMilken) [reply uttut/pipeline/edit/base.py]: @absolutelyNoWarranty

Comment (SoluMilken) [reply uttut/pipeline/edit/base.py]: A replacement is a data structure that records the procedure of replacing a slice of original sequence with 
new_value.

File: uttut/pipeline/edit/lst2lst.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+from .utils import (
+    _transform_sequence,
+    _gen_inverse_replacement_group,
+)
+
+
+def apply(input_lst: list, replacement_group: ReplacementGroup) -> List[str]:
+    output: List[list]
+    n_replacement = len(replacement_group)
+    output = [[]] * (2 * n_replacement + 1)
```
Comment (GBLin5566) [inline uttut/pipeline/edit/lst2lst.py]:
```python
output: List[list] = [[]] * (2 * n_replacement + 1) 
```

File: uttut/pipeline/edit/lst2lst.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+from .replacement import ReplacementGroup
+from .utils import (
+    _transform_sequence,
+    _gen_inverse_replacement_group,
+)
+
+
+def apply(input_lst: list, replacement_group: ReplacementGroup) -> List[str]:
+    output: List[list]
```
Comment (GBLin5566) [inline uttut/pipeline/edit/lst2lst.py]: Remove this line ?

File: uttut/pipeline/edit/base.py (hunk 1 above)
Comment (absolutelyNoWarranty) [reply uttut/pipeline/edit/base.py]: A `Replacement` stores the data for substitution of values in a sequence. It represents the replacing a continuous slice of the sequence with the items of `new_values`. (The semantics are similar to Python's built-in slice assignment feature https://docs.python.org/3/reference/simple_stmts.html#assignment-statements , but more restrictive in that negative integers are not allowed for `start` and `end`. )

**Pull request merged**